inbm/dispatcher-agent/dispatcher/sota/sota.py

```diff
@@ -165,11 +167,16 @@ def calculate_and_execute_sota_upgrade(self, repo: IRepo) -> List:
                                              self._install_check_service)
             if self._repo_type == REMOTE_SOURCE:
                 logger.debug(f"Remote repo URI: {self._uri}")
-                if self._uri is None:
-                    cmd_list = self.installer.update_remote_source(None, repo)
+                if isinstance(self.factory, TiberOSBasedSotaOs):
```

Comment:
the factory pattern forbids examining the type here--would be better simply to pass in all arguments that might be needed and allow the individual types to ignore the ones they don't need

also (I may have mentioned this elsewhere), `signature` would be clearer as `signature_uri`

Reply:
Type examination removed. Added if else statement.

                if self._uri is not None and self._signature is not None:
                    cmd_list = self.installer.update_remote_source(canonicalize_uri(self._uri),
                                                                   canonicalize_uri(self._signature), repo)
                elif self._uri is None:
                    cmd_list = self.installer.update_remote_source(None, None, repo)
                else:
                    cmd_list = self.installer.update_remote_source(
                        canonicalize_uri(self._uri), None, repo)